File: java/aoc_utils/geometry/Position3D.java
package aoc_utils.geometry;

public class Position3D<T> {
    public T x;
    public T y;
    public T z;

    public Position3D(T x, T y, T z) {
        this.x = x;
        this.y = y;
	this.z = z;
    }
}

File: swift/utils/AOC_Utils/Sources/AOC_Utils/Geometry/Position3D.swift
extension Geometry {
    public struct Position3D<T: Numeric & Hashable>: Hashable {
        public var x: T
        public var y: T
        public var z: T

        public init(_ x: T, _ y: T, _ z: T) {
            self.x = x
            self.y = y
            self.z = z
        }
    }
}
